OMG.Classroom/DataAccessLayer.Repositories.Tests/AssignmentRepositoryTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿using AutoMapper;
+using DataAccessLayer.Entities;
+using DataAccessLayer.Mappers;
+using DataAccessLayer.Repositories.AssignmentRepositories;
+using DataAccessLayer.Repositories.RoleRepositories;
+using DataAccessLayer.Seeder;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace DataAccessLayer.Repositories.Tests
+{
+    public class AssignmentRepositoryTests : IDisposable
+    {
+        private readonly DbContextOptions<ClassroomDbContext> dbContextOptions;
+        private readonly ClassroomDbContext context;
+        private readonly IMapper _mapper;
+        private readonly AssignmentRepository repository;
+
+        public async void Dispose()
+        {
+            await context.Database.EnsureDeletedAsync();
+        }
+
+        public AssignmentRepositoryTests()
+        {
+            dbContextOptions = new DbContextOptionsBuilder<ClassroomDbContext>()
+                .UseInMemoryDatabase(databaseName: "reposdb").Options;
+
+            if (_mapper == null)
+            {
+                var mappingConfig = new MapperConfiguration(mc =>
+                {
+                    mc.AddProfile(new MapperProfile());
+                });
+                IMapper mapper = mappingConfig.CreateMapper();
+                _mapper = mapper;
+            }
+
+            context = new ClassroomDbContext(dbContextOptions);
+            repository = new AssignmentRepository(context);
+        }
+
+        public async Task SeedDataAsync(ClassroomDbContext context)
+        {
+            await Seed.SeedRolesAsync(context);
+        }
+
+        [Fact]
+        public async Task CanAddAssignment()
+        {
+            var newAssignment = new Assignment();
+            newAssignment.Name = "new role";
```

Comment:
Is Name the only non-navigation property in Assignment? Consider this for the other classes as well